x/staking/types/historical_info.go

```diff
@@ -54,3 +57,14 @@ func ValidateBasic(hi HistoricalInfo) error {
 
 	return nil
 }
+
+// UnpackInterfaces implements UnpackInterfacesMessage.UnpackInterfaces
+func (hi *HistoricalInfo) UnpackInterfaces(unpacker codectypes.AnyUnpacker) error {
+	for i := range hi.Valset {
+		var pk crypto.PubKey
+		if err := unpacker.UnpackAny(hi.Valset[i].ConsensusPubkey, &pk); err != nil {
```

Comment:
Since you implemented `UnpackInterfaces` on Validator, here, it could just be:

```go
for _, val := range hi.Valset {
    err := val.UnpackInterfaces(unpacker)
    if err != nil {
        return err
    }
}
```